test/unit/models/classes/xml/XmlEditorTest.php

```diff
@@ -65,10 +74,12 @@ public function setUp(): void
             ->with($this->testResourceMock)
             ->willReturn($this->xmlDoc);
         $this->featureFlagCheckerMock = $this->createMock(FeatureFlagChecker::class);
+        $this->userSettingsService = $this->createMock(UserSettingsService::class);
```

Comment:
chore: Lets rely on interface for open/closed implementation best practices

```suggestion
        $this->userSettingsService = $this->createMock(UserSettingsServiceInterface::class);
```